### PR TITLE
DEG-190: Site studio integration with DST.

### PR DESCRIPTION
**Pull request by singhprerana**
A similar PR may already be submitted!
Please search among the [Pull request](https://github.com/acquia-pso/dst-entity-generate/pulls) before creating one.

Thanks for submitting a pull request! Please provide enough information so that others can review your pull request:

For more information, see the [CONTRIBUTING guide](https://github.com/acquia-pso/dst-entity-generate/blob/master/CONTRIBUTING.md).

**Summary**
This PR contains code block to integrate site studio website settings with DEG module.

This PR fixes/implements the following features:

* Integration with site studio color, font stack and font library.
* Commands to generate/update all entities
```
drush deg:ss 
drush deg:ss  —update
```

- Commands to generate specific entity
```
drush deg:ss  —type color
drush deg:ss  —type font_stack
drush deg:ss  —type font_library
```

- Commands to update specific entity
```
drush deg:ss  —update —type color
drush deg:ss  —update —type font_stack
drush deg:ss  —update —type 
```
The reference DST sheet:

image

### CODE REVIEW

File: src/Commands/SiteStudioSettings.php [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    foreach ($data as $item) {
+      // To allow existing entity update.
+      if (!$this->validateMachineName($item['machine_name'])) {
+        continue;
+      }
+      // Type value will distinguish between the cohesion entities.
+      // Command with param -> type = param.
+      // Command with no param -> type value from sheet data to be considered.
+      $type = $param ?: strtolower($item['type']);
+      switch ($type) {
```
Comment (vyasamit2007) [inline src/Commands/SiteStudioSettings.php:136]: **Avoid code duplicacy in switch cases**
Avoid duplicate code statements in switch case.

File: src/Commands/SiteStudioSettings.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    $rgba_color = $this->hex2rgba($item['metadata'], 1);
+    if (array_key_exists('tag', $item) && $item['tag']) {
+      $tag = [$item['tag']];
+    } else {
+      $tag = [];
+    }
+    return [
+      'link' => true,
+      'value' => [
+        'value' => [
```
Comment (vyasamit2007) [inline src/Commands/SiteStudioSettings.php:235]: is this duplicate `value` works fine OR this is the format required?

**Review by vyasamit2007 — CHANGES_REQUESTED** [review]

File: src/Commands/SiteStudioSettings.php (hunk 2 above)
Comment (singhprerana) [reply src/Commands/SiteStudioSettings.php:235]: This is the format of json_values for cohesion color.

File: src/Commands/SiteStudioSettings.php (hunk 1 above)
Comment (singhprerana) [reply src/Commands/SiteStudioSettings.php:136]: Each switch case makes a call to different methods based on cohesion types -> font stack, font library, color.

**Review by vyasamit2007 — APPROVED** [review]

**Pull request merged**